trip_app/trips/views.py

```diff
@@ -1,45 +1,91 @@
-from core.utils import str_to_geopoint
 from django.contrib.gis.db.models.functions import Distance
 from django.db.models import F
-from rest_framework.generics import ListCreateAPIView
+from rest_framework import status
+from rest_framework.decorators import action
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
 
-from .models import Trip
-from .serializers import TripSerializer
+from core.utils import str_to_geopoint
+from trips.models import Trip, TripRequest
+from trips.permissions import IsTripDriverOrAdmin
+from trips.serializers import TripSerializer, TripRequestSerializer
 
 
-class TripListCreateApiView(ListCreateAPIView):
+class TripViewSet(ModelViewSet):
     serializer_class = TripSerializer
-    permission_classes = [IsAuthenticated, ]
     pagination_class = PageNumberPagination
     pagination_class.page_size = 5
 
     def perform_create(self, serializer):
         serializer.save(driver=self.request.user)
 
+    def get_permissions(self):
+        if self.action in ["update", "destroy", "partial_update"]:
+            self.permission_classes = [IsTripDriverOrAdmin]
+        else:
+            self.permission_classes = [IsAuthenticated]
+        return super().get_permissions()
+
     def get_queryset(self):
         queryset = Trip.objects.all()
         query_params = self.request.query_params
 
         time1 = query_params.get("time1")
         time2 = query_params.get("time2")
-        sp = query_params.get("sp")
-        dp = query_params.get("dp")
+        start_point = query_params.get("sp")
```

Comment:
I don't get why you sending coordinates like this `coord1 coord2`, but not `coord1, coord2`. It is easier to parse values then from regular string, and then in `str_to_geopoint` you do not need any regex to parse something. Just checked with gmaps, and they use comma between `x,y` coordinate. Explain me why you chose such style if there is any reason for this 🙃

Reply:
Well... initially I wasn't sure what are the rules and formats therefore decided to use re here to expand the range of incoming formats of data. 
I expected the clients would send us coordinates in string formats such as '23.4 23.5', '[23.4 23.5]', '(23.4 23.5)'.. 
Not it looks weird though, but I'm planing to address it in next PR and add drf-gis library that contains proper serializers to be able to get rid of `str_to_geopoint` and in general clear things up.